easybuild/tools/docs.py

```diff
@@ -744,28 +744,65 @@ def list_toolchains_rst(tcs):
     """ Returns overview of all toolchains in rst format """
     title = "List of known toolchains"
 
-    # figure out column names
-    table_titles = ['name', 'compiler', 'MPI']
-    for tc in tcs.values():
-        table_titles.extend(tc.keys())
+    # Specify the column names for the table
+    table_titles = ['name', 'COMPILER', 'MPI', 'linalg', 'FFT']
 
+    # Set up column name : display name pairs
     col_names = {
-        'COMPILER_CUDA': 'CUDA compiler',
-        'SCALAPACK': 'ScaLAPACK',
+        'name': 'Name',
+        'compiler': 'Compiler(s)',
+        'linalg': "Linear algebra",
     }
 
-    table_titles = nub(table_titles)
+    # Create sorted list of toolchain names
+    sorted_tc_names = sorted(tcs.keys(), key=str.lower)
 
-    table_values = [[] for i in range(len(table_titles))]
-    table_values[0] = ['**%s**' % tcname for tcname in tcs.keys()]
+    # Create text placeholder to use for missing entries.
+    # Not good documentation style to italicize the parentheses enclosing
+    # italic text unless surrouning text is also italic.
+    none_txt = '(*none*)'
 
-    for idx in range(1, len(table_titles)):
-        for tc in tcs.values():
-            table_values[idx].append(', '.join(tc.get(table_titles[idx].upper(), [])))
+    # Initialize an empty list of lists for the table data
+    table_values = [[] for i in range(len(table_titles))]
 
+    for col_id, col_name in enumerate(table_titles):
+        if col_name == 'name':
+            # toolchain names column gets bold face entry
+            table_values[col_id] = ['**%s**' % tcname for tcname in sorted_tc_names]
+        else:
+            for tc_name in sorted_tc_names:
+                tc = tcs[tc_name]
+                # Cray is a special snowflake
+                if tc_name.find('Cray') >= 0:
+                    if col_name == 'COMPILER':
+                        entry = 'PrgEnv-cray'
+                    elif col_name == 'MPI':
+                        entry = 'cray-mpich'
+                    elif col_name == 'linalg':
+                        entry = 'cray-libsci'
+                # Combine the linear algebra libraries into a single column
+                elif col_name == 'linalg':
+                    linalg = []
+                    for col in ['BLAS', 'LAPACK', 'SCALAPACK']:
+                        linalg.extend(tc.get(col, [none_txt]))
+                    # Doing this by iterating over the columns results in multiple 'none'
+                    # entries, so reduce those to one.
+                    linalg = nub(linalg)
+                    entry = ', '.join(linalg)
+                else:
+                    # for other columns, we can grab the values via 'tc'
+                    # key = col_name
+                    entry = ', '.join(tc.get(col_name, [none_txt]))
```

Comment:
use `col_name.upper()` as key (important w.r.t. changing `'COMPILER'` to `'compiler'` above):

```python
key = col_name.upper()
entry = ', '.join(tc.get(key, [none_txt]))
```

or

```python
entry = ', '.join(tc.get(col_name.upper(), [none_txt]))

Reply:
Your first comment wants to change `COMPILER` to `compiler`.  I think this is a matter of consistency throughout the block.  Whatever is used in `table_titles` should be used throughout for things to work properly.  The column heading was coming out `COMPILER` because I initially failed to make it a matching 'COMPILER' in `col_names`.  That is fixed, and it is `COMPILER` everywhere, which works.

Reply:
It sort of works by accident now, without the `.upper()`, because `name` and `linalg` are handled separate as special cases...

If someone would change `table_titles` (to use `'compiler'` rather than `'COMPILER'` for example ;) ), this would fail with a `.upper()` (and having the `.upper()` doesn't hurt).